usaspending_api/recipient/management/sql/recipient_lookup/030_sam_parents.sql

```diff
@@ -10,7 +10,7 @@ INSERT INTO public.temporary_restock_recipient_lookup (
 )
 SELECT
   DISTINCT ON (ultimate_parent_unique_ide, ultimate_parent_legal_enti)
-  MD5(UPPER(CONCAT('duns-', ultimate_parent_unique_ide)))::uuid AS recipient_hash,
+  MD5(COALESCE(UPPER(CONCAT('uei-', ultimate_parent_uei)), UPPER(CONCAT('duns-', ultimate_parent_unique_ide))))::uuid AS recipient_hash,
   UPPER(ultimate_parent_legal_enti) AS legal_business_name,
   ultimate_parent_unique_ide AS duns,
   'sam-parent' as source,
```

Comment:
Also, it's not a bug in this `030` step, but there's a related bug in the `070` step which would be good to fix here if possible:

In `070 Adding Recipient records from SAM parent data without a name`, basically need to add the `AND ultimate_parent_legal_enti IS NULL` to the end of the WHERE clause. 
```
    -- BUG fixed here: originally wasn't pinning ultimate_parent_legal_enti to NULL, which is reintroducing data where both are provided
    WHERE ultimate_parent_unique_ide IS NOT NULL AND ultimate_parent_legal_enti IS NULL
```